flume-ng-sinks/flume-dataset-sink/src/main/java/org/apache/flume/sink/kite/DatasetSink.java

```diff
@@ -48,144 +39,157 @@
-import org.kitesdk.data.DatasetException;
+import org.kitesdk.data.DatasetIOException;
+import org.kitesdk.data.DatasetNotFoundException;
 import org.kitesdk.data.DatasetWriter;
+import org.kitesdk.data.DatasetWriterException;
 import org.kitesdk.data.Datasets;
 import org.kitesdk.data.View;
 import org.kitesdk.data.spi.Registration;
-import org.kitesdk.data.spi.URIBuilder;
+import org.kitesdk.data.URIBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static org.apache.flume.sink.kite.DatasetSinkConstants.*;
+
 /**
- * Experimental sink that writes events to a Kite Dataset. This sink will
- * deserialize the body of each incoming event and store the resulting record
- * in a Kite Dataset. It determines target Dataset by opening a repository URI,
- * {@code kite.repo.uri}, and loading a Dataset by name,
- * {@code kite.dataset.name}.
+ * Sink that writes events to a Kite Dataset. This sink will parse the body of
+ * each incoming event and store the resulting entity in a Kite Dataset. It
+ * determines datasetUri Dataset by opening a dataset URI
+ * {@code kite.dataset.uri} or opening a repository URI, {@code kite.repo.uri},
+ * and loading a Dataset by name, {@code kite.dataset.name}.
  */
 public class DatasetSink extends AbstractSink implements Configurable {
 
   private static final Logger LOG = LoggerFactory.getLogger(DatasetSink.class);
 
-  static Configuration conf = new Configuration();
+  private Context context = null;
+  private UserGroupInformation login = null;
 
   private String datasetName = null;
-  private long batchSize = DatasetSinkConstants.DEFAULT_BATCH_SIZE;
-
-  private URI target = null;
-  private Schema targetSchema = null;
+  private URI datasetUri = null;
+  private Schema datasetSchema = null;
   private DatasetWriter<GenericRecord> writer = null;
-  private UserGroupInformation login = null;
-  private SinkCounter counter = null;
 
-  // for rolling files at a given interval
-  private int rollIntervalS = DatasetSinkConstants.DEFAULT_ROLL_INTERVAL;
+  /**
+   * The number of events to process as a single batch.
+   */
+  private long batchSize = DEFAULT_BATCH_SIZE;
+
+  /**
+   * The number of seconds to wait before rolling a writer.
+   */
+  private int rollIntervalS = DEFAULT_ROLL_INTERVAL;
+
+  /**
+   * Flag that says if Flume should commit on every batch.
+   */
+  private boolean commitOnBatch = DEFAULT_COMMIT_ON_BATCH;
+
+  /**
+   * The last time the writer rolled.
+   */
   private long lastRolledMs = 0l;
 
-  // for working with avro serialized records
-  private GenericRecord datum = null;
+  /**
+   * The raw number of bytes parsed.
+   */
+  private long bytesParsed = 0l;
+
+  /**
+   * A class for parsing Kite entities from Flume Events.
+   */
+  private EntityParser<GenericRecord> parser = null;
+
+  /**
+   * A class implementing a failure newPolicy for events that had a
+ non-recoverable error during processing.
+   */
+  private FailurePolicy policy = null;
```

Comment:
It would be more clear in the code that uses this if it were named `failurePolicy`.